File: src/utils/logger.ts
const PREFIX = '[promptcache]'

export function createLogger(enabled: boolean = false) {
  return {
    debug: (...args: unknown[]) => {
      if (enabled) console.log(PREFIX, ...args)
    },
    info: (...args: unknown[]) => {
      if (enabled) console.info(PREFIX, ...args)
    },
    warn: (...args: unknown[]) => {
      console.warn(PREFIX, ...args)
    },
  }
}

File: python/cachellm/utils/logger.py
import logging

_logger = logging.getLogger("cachellm")

PREFIX = "[cachellm]"


class Logger:
    def __init__(self, enabled: bool = False):
        self._enabled = enabled

    def debug(self, *args: object) -> None:
        if self._enabled:
            print(PREFIX, *args)

    def info(self, *args: object) -> None:
        if self._enabled:
            print(PREFIX, *args)

    def warn(self, *args: object) -> None:
        print(PREFIX, *args)


def create_logger(enabled: bool = False) -> Logger:
    return Logger(enabled)
